# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1425168468 18000
#      Sat Feb 28 19:07:48 2015 -0500
# Node ID 2350fc8ba960eebf7aeffbc37ddb275a5e64058e
# Parent  7575048a555bf9c4d77cbc905a06bcad74b94d04
build: Fix build failure on missing oct-parse.h (bug #44391)

* libinterp/Makefile.am (GENERATED_MAKE_BUILTINS_INCS): Include
parse-tree/oct-parse.h in the list.
* libinterp/parse-tree/module.mk (parse-tree/oct-parse.h): Depend on
parse-tree/oct-parse.cc.

diff --git a/libinterp/Makefile.am b/libinterp/Makefile.am
--- a/libinterp/Makefile.am
+++ b/libinterp/Makefile.am
@@ -47,16 +47,17 @@ octlib_LTLIBRARIES = liboctinterp.la
 GENERATED_MAKE_BUILTINS_INCS = \
   corefcn/defaults.h \
   corefcn/graphics-props.cc \
   corefcn/graphics.h \
   corefcn/mxarray.h \
   oct-conf-features.h \
   oct-conf.h \
   parse-tree/oct-gperf.h \
+  parse-tree/oct-parse.h \
   version.h
 
 BUILT_SOURCES = \
   $(GENERATED_MAKE_BUILTINS_INCS) \
   builtin-defun-decls.h \
   builtins.cc \
   corefcn/oct-errno.cc \
   corefcn/oct-tex-lexer.cc \
diff --git a/libinterp/parse-tree/module.mk b/libinterp/parse-tree/module.mk
--- a/libinterp/parse-tree/module.mk
+++ b/libinterp/parse-tree/module.mk
@@ -85,16 +85,18 @@ PARSE_TREE_SRC = \
 ## oct-gperf.h file.
 parse-tree/oct-gperf.h: parse-tree/octave.gperf
 	$(AM_V_GEN)rm -f $@-t $@t1 $@ && \
 	$(GPERF) -t -C -D -G -L C++ -Z octave_kw_hash $< > $@-t1 && \
 	$(SED) 's,lookup\[,gperf_lookup[,' < $@-t1 > $@-t && \
 	mv $@-t $@ && \
 	rm -f $@-t1
 
+parse-tree/oct-parse.h: parse-tree/oct-parse.cc
+
 parse-tree/oct-parse.yy: parse-tree/oct-parse.in.yy
 	$(AM_V_GEN)$(call subst-bison-api-decls,octave_)
 
 noinst_LTLIBRARIES += \
   parse-tree/libparse-tree.la \
   parse-tree/libparser.la
 
 parse_tree_libparse_tree_la_SOURCES = $(PARSE_TREE_SRC)
